### PR TITLE
Make packageName non-optional

### PR DESCRIPTION
**Pull request by tudor-da**

### CODE REVIEW

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala [hunk 1]
```diff
@@ -1954,11 +1954,7 @@ final class SBuiltinTestHelpers(majorLanguageVersion: LanguageMajorVersion) {
     """
 
   val txVersion = TransactionVersion.assignNodeVersion(pkg.languageVersion)
-  val pkgName =
-    if (txVersion < TransactionVersion.minUpgrade)
```
Comment (remyhaemmerle-da) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/SBuiltinTest.scala:1958]: Can you remove `TransactionVersion.minUpgrade` ?

File: daml-lf/transaction-test-lib/src/main/scala/lf/transaction/test/TestNodeBuilder.scala [hunk 2]
```diff
@@ -33,8 +31,7 @@ trait TestNodeBuilder {
       signatories: Set[Party],
       observers: Set[Party] = Set.empty,
       key: CreateKey = CreateKey.NoKey,
-      // TODO: https://github.com/digital-asset/daml/issues/17995
-      //  review if we should really provide a defaul package name.
+      // TODO(Tudor): Remove
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction-test-lib/src/main/scala/lf/transaction/test/TestNodeBuilder.scala:34]:
```suggestion
      // TODO https://github.com/digital-asset/daml/issues/17995
```

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala [hunk 3]
```diff
@@ -47,9 +48,8 @@ class ContractStateMachineSpec extends AnyWordSpec with Matchers with TableDrive
   val aliceS: Set[Ref.Party] = Set(alice)
   val templateId: Ref.TypeConName = "Template:Id"
   val choiceId: Ref.ChoiceName = "Choice"
-  val pkgName = Some(Ref.PackageName.assertFromString("package-name"))
+  val pkgName: IdString.PackageName = Ref.PackageName.assertFromString("package-name")
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala:51]:
```suggestion
  val pkgName: Ref.PackageName = Ref.PackageName.assertFromString("package-name")
```

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala [hunk 4]
```diff
@@ -5,6 +5,7 @@ package com.daml
 package lf
 package transaction
 
+import com.daml.lf.data.Ref.IdString
```
Comment (remyhaemmerle-da) [inline daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala:8]:
```suggestion
```

File: daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala (hunk 3 above)
Comment (remyhaemmerle-da) [reply daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala:51]: stupid IJ

**Review by remyhaemmerle-da — APPROVED** [review]
Nice. Thanks a lot.

File: daml-lf/transaction-test-lib/src/main/scala/lf/transaction/test/TestNodeBuilder.scala (hunk 2 above)
Comment (tudor-da) [reply daml-lf/transaction-test-lib/src/main/scala/lf/transaction/test/TestNodeBuilder.scala]: Done

**Pull request merged**